storage/reader.go

```diff
@@ -349,6 +485,25 @@ func setConditionsHeaders(headers http.Header, conds *Conditions) error {
 	return nil
 }
 
+func setRequestConditions(req *storagepb.ReadObjectRequest, conds *Conditions) {
```

Comment:
Let's separate out these utility functions for gRPC and put them at the bottom (with comments marking them as non-production at this point)

Reply:
I've moved everything gRPC related to a block at the bottom.